### PR TITLE
MULE-11220: Review the HttpEntity interface and its Implementations.

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

**Comment by afelisatti** [comment]
Tests currently running on CI.

File: modules/http-api/src/main/java/org/mule/runtime/http/api/domain/entity/HttpEntity.java [hunk 1]
```diff
@@ -6,11 +6,70 @@
...
+   * Provides the entity's content as a stream. All streaming entities should provide their stream.
+   *
+   * @return an {@code InputStream} representing this entity's content or {@code null} if such representation is not possible
+   */
+  InputStream getContent();
+
+  /**
+   * Provides the entity's content as bytes. If the entity is stream based, then the stream will be consumed as a consequence.
+   *
+   * @return a byte array representing this entity's content or {@code null} if such representation is not possible
```
Comment (elrodro83) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/entity/HttpEntity.java]: `or {@code null} if such representation is not possible` -> isn't the implementation returning an emty array in this case?

Comment (afelisatti) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/entity/HttpEntity.java]: I'll check. If I remember correctly the only entity which did not allow stream not byte representation was MultipartHttpEntity. The rest, even EmptyHttpEntity, have ways of doing so.

Comment (afelisatti) [inline modules/http-api/src/main/java/org/mule/runtime/http/api/domain/entity/HttpEntity.java]: Yes, MultipartHttpEntity signals with null that it cannot handle that representation.

**Pull request merged**